[ML][TEST] Fix limits in AutodetectMemoryLimitIT

edsavage opened the pull request:
Re-enable muted tests and accommodate recent backend changes
that result in higher memory usage being reported for a job
at the start of its life-cycle

Relates elastic/ml-cpp#486
Fixes #42207

edsavage committed:
[ML][TEST] Fix limits in AutodetectMemoryLimitIT

Re-enable muted tests and accommodate recent backend changes
that result in higher memory usage being reported for a job
at the start of its life-cycle

Fixes #42207

elasticmachine commented:
Pinging @elastic/ml-core

edsavage committed:
Merge branch 'master' of github.com:elastic/elasticsearch into test-fix/autodetect_memory_limits

droberts195 commented on x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/AutodetectMemoryLimitIT.java:
```diff
@@ -125,12 +124,11 @@ public void testTooManyByFields() throws Exception {
         GetJobsStatsAction.Response.JobStats jobStats = getJobStats(job.getId()).get(0);
         ModelSizeStats modelSizeStats = jobStats.getModelSizeStats();
-        assertThat(modelSizeStats.getModelBytes(), lessThan(31500000L));
+        assertThat(modelSizeStats.getModelBytes(), lessThan(33000000L));
```
I would make it `35000000L` to allow room for platform variations.

droberts195 commented on x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/AutodetectMemoryLimitIT.java:
```diff
@@ -175,12 +173,11 @@ public void testTooManyByAndOverFields() throws Exception {
         GetJobsStatsAction.Response.JobStats jobStats = getJobStats(job.getId()).get(0);
         ModelSizeStats modelSizeStats = jobStats.getModelSizeStats();
-        assertThat(modelSizeStats.getModelBytes(), lessThan(31500000L));
+        assertThat(modelSizeStats.getModelBytes(), lessThan(31800000L));
```
Maybe `33000000L` in case Mac or Windows uses more.

droberts195 commented on x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/AutodetectMemoryLimitIT.java:
```diff
@@ -226,7 +223,7 @@ public void testManyDistinctOverFields() throws Exception {
         GetJobsStatsAction.Response.JobStats jobStats = getJobStats(job.getId()).get(0);
         ModelSizeStats modelSizeStats = jobStats.getModelSizeStats();
-        assertThat(modelSizeStats.getModelBytes(), lessThan(116000000L));
+        assertThat(modelSizeStats.getModelBytes(), lessThan(116100000L));
```
How about `117000000L` to allow a bit more wiggle room?

edsavage committed:
Further loosening the upper bounds

Increasing the high memory limit even further to avoid disappointment
with CI builds on other platforms, future changes etc.

edsavage committed:
Merge branch 'master' of github.com:elastic/elasticsearch into test-fix/autodetect_memory_limits

edsavage commented:
Jenkins run elasticsearch-ci/packaging-sample

droberts195 reviewed (approved):
LGTM

edsavage commented:
Jenkins run elasticsearch-ci/1

edsavage committed:
Merge branch 'master' of github.com:elastic/elasticsearch into test-fix/autodetect_memory_limits

edsavage commented:
Jenkins run elasticsearch-ci/2

edsavage committed:
Merge branch 'master' of github.com:elastic/elasticsearch into test-fix/autodetect_memory_limits

edsavage committed:
Merge branch 'master' of github.com:elastic/elasticsearch into test-fix/autodetect_memory_limits
